3rdparty/musl/README.md

```diff
@@ -4,7 +4,7 @@ musl libc
 This directory contains the musl libc library and the accompanying libc-test test
 suite. The structure of the directory is as follows.
 
-- musl/ (tag: `v1.1.21`)
+- musl/ (tag: `v1.1.23`)
```

Comment:
Shouldn't we be upgrading to v1.2.2 given that v1.1.x is [end-of-life](https://musl.libc.org/releases.html)?

Reply:
@CodeMonkeyLeet @anakrish do we expect breaking changes for v1.2.2? If you expect major differences over current version, would prefer to keep to this plan since we will release next week and have a short window to  test. We can move to 1.2.2 post release

Reply:
Given that one of the reasons for the MUSL upgrade is to address known vulnerabilities before the release, I would suggest at least trying the upgrade to 1.2.2 directly. Based on the release history, the primary breaking change was the migration of 32-bit architectures to using 64-bit `time_t` which should not affect OE since we only build on 64-bit.

Reply:
Thank you @CodeMonkeyLeet, @anakrish . I am working on upgrading to v1.2.2. There are quite a few additions and file structure changes that I am addressing. I was thinking of adding that upgrade it as a separate PR, so that this version can get in for the release (at the least) and v1.2.2 in next couple days. Is that okay?

Reply:
I took a quick look at upgrading to v1.2.2. It breaks our assumptions around pthread, some match functions and many more incompatibilities.
I don't think we can upgrade to v1.2.2 in a week. Therefore I'm OK with this PR to upgrade to v1.1.23 and am removing my blocker.

However, I'd suggest that in future we do upgrades early in the release cycle so that we get enough time to upgrade to the correct version as well as do necessary testing.